SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -1150,49 +1153,103 @@ void CNEMOEulerSolver::Source_Residual(CGeometry *geometry, CSolver **solver_con
-      //      if (Jacobian_i[iVar][jVar] != Jacobian_i[iVar][jVar]) err = true;
+      if (viscous) {
 
-      /*--- Apply the update to the linear system ---*/
-      if (!err) {
-        LinSysRes.AddBlock(iPoint, residual);
-        //if (implicit)
-        //  Jacobian.AddBlock(iPoint, iPoint, Jacobian_i);
+        for (iPoint = 0; iPoint < nPoint; iPoint++) {
+
+          su2double yCoord          = geometry->nodes->GetCoord(iPoint, 1);
+          su2double yVelocity       = nodes->GetVelocity(iPoint,1);
+          su2double xVelocity       = nodes->GetVelocity(iPoint,0);
+          su2double Total_Viscosity = nodes->GetLaminarViscosity(iPoint) + nodes->GetEddyViscosity(iPoint);
+
+          if (yCoord > EPS){
+            su2double nu_v_on_y = Total_Viscosity*yVelocity/yCoord;
+            nodes->SetAuxVar(iPoint, 0, nu_v_on_y);
+            nodes->SetAuxVar(iPoint, 1, nu_v_on_y*yVelocity);
+            nodes->SetAuxVar(iPoint, 2, nu_v_on_y*xVelocity);
+          }
+        }
+
+        /*--- Compute the auxiliary variable gradient with GG or WLS. ---*/
+        if (config->GetKind_Gradient_Method() == GREEN_GAUSS) {
+          SetAuxVar_Gradient_GG(geometry, config);
+        }
+        if (config->GetKind_Gradient_Method() == WEIGHTED_LEAST_SQUARES) {
+          SetAuxVar_Gradient_LS(geometry, config);
+        }
       }
-      else
-        eAxi_local++;
-    }
 
-    if(!monoatomic){
-      if(!frozen){
-        /*--- Compute the non-equilibrium chemistry ---*/
-        auto residual = numerics->ComputeChemistry(config);
+      /*--- loop over points ---*/
+      SU2_OMP_FOR_DYN(omp_chunk_size)
+      for (iPoint = 0; iPoint < nPointDomain; iPoint++) {
 
-        /*--- Check for errors before applying source to the linear system ---*/
-        err = false;
-        for (iVar = 0; iVar < nVar; iVar++)
-          if (residual[iVar] != residual[iVar]) err = true;
-        //if (implicit)
-        //  for (iVar = 0; iVar < nVar; iVar++)
-        //    for (jVar = 0; jVar < nVar; jVar++)
-        //      if (Jacobian_i[iVar][jVar] != Jacobian_i[iVar][jVar]) err = true;
-
-        /*--- Apply the chemical sources to the linear system ---*/
-        if (!err) {
-          LinSysRes.SubtractBlock(iPoint, residual);
-          //if (implicit)
-          //  Jacobian.SubtractBlock(iPoint, iPoint, Jacobian_i);
-        } else
-          eChm_local++;
+        /*--- Set solution  ---*/
+        numerics->SetConservative(nodes->GetSolution(iPoint), nodes->GetSolution(iPoint));
+
+        /*--- Set control volume ---*/
+        numerics->SetVolume(geometry->nodes->GetVolume(iPoint));
+
+        /*--- Set y coordinate ---*/
+        numerics->SetCoord(geometry->nodes->GetCoord(iPoint), geometry->nodes->GetCoord(iPoint));
+
+        /*--- Set primitive variables for viscous terms and/or generalised source ---*/
+        numerics->SetPrimitive(nodes->GetPrimitive(iPoint), nodes->GetPrimitive(iPoint));
+
+        if (viscous) {
```

Comment:
maybe add a comment here

Reply:
Is this what you mean by a comment here?

Reply:
Yup!